pkg/apis/topology/v1alpha1/hypernode_types.go

```diff
@@ -0,0 +1,160 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +genclient:nonNamespaced
+// +kubebuilder:resource:path=hypernodes,shortName=hn,scope=Cluster
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Tier",type=string,JSONPath=`.spec.tier`
+// +kubebuilder:printcolumn:name="NodeCount",type=integer,JSONPath=`.status.nodeCount`
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+
+// HyperNode represents a collection of nodes sharing similar network topology or performance characteristics.
+type HyperNode struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired configuration of the HyperNode.
+	// +optional
+	Spec HyperNodeSpec `json:"spec"`
+
+	// Status provides the current state of the HyperNode.
+	// +optional
+	Status HyperNodeStatus `json:"status,omitempty"`
+}
+
+// MemberType represents the member type, valid values are "Node" and "HyperNode".
+// +kubebuilder:validation:Enum=Node;HyperNode
+type MemberType string
+
+const (
+	// MemberTypeNode means the member type is a node.
+	MemberTypeNode MemberType = "Node"
+	// memberTypeHyperNode means the member type is a hyperNode.
+	memberTypeHyperNode MemberType = "HyperNode"
+)
+
+// HyperNodeSpec defines the desired state of a HyperNode.
+type HyperNodeSpec struct {
+	// Tier categorizes the performance level of the HyperNode.
+	// +required
+	Tier string `json:"tier,omitempty"`
+
+	// Members defines a list of node groups or individual nodes included in the HyperNode.
+	// +optional
+	Members []MemberSpec `json:"members,omitempty"`
+}
+
+// MemberSpec represents a specific node or a hyperNodes in the hyperNode.
+type MemberSpec struct {
+	// Type specifies the member type.
+	// +required
+	Type MemberType `json:"type,omitempty"`
+
+	// Selector defines the selection rules for this member.
+	// +optional
+	Selector MemberSelector `json:"selector,omitempty"`
+}
+
+// MemberSelectorType specifies the selection method, valid values are "Exact" and "Regex".
+// +kubebuilder:validation:Enum=Exact;Regex
+type MemberSelectorType string
+
+const (
+	// ExactMatchMemberSelectorType matches nodes exactly by name.
+	ExactMatchMemberSelectorType MemberSelectorType = "Exact"
+
+	// RegexMatchMemberSelectorType matches nodes based on a regular expression.
+	RegexMatchMemberSelectorType MemberSelectorType = "Regex"
+)
+
+// MemberSelector defines the criteria for selecting nodes.
+//
+// Example for Exact match:
+//
+//	members:
+//	- selector:
+//	    type: Exact
+//	    exactMatch:
```

Comment:
the `exactMatch` and `regexMatch` have let tell volcano which expression to use. What's the intention of `type`?

Reply:
It's more clear to give a more detail expression.